packages/components/src/components/TableComponent/TableComponent.stories.js

```diff
@@ -0,0 +1,156 @@
+      }
+    ];
+
+    return (
+      <TableComponent
+        title={text('Title', 'Resource Name')}
+        dataRows={dataRows}
+        dataHeaders={[
+          { key: 'name', header: 'Name' },
+          { key: 'namespace', header: 'Namespace' },
+          { key: 'date', header: 'Date Modified' }
+        ]}
+        selectedNamespace="*"
+        loading={boolean('loading', false)}
+        handleNew={action('handleNew')}
+      />
+    );
+  })
+  .add('table with one row, Add & Delete button and checkboxes', () => {
+    const dataRows = [
+      {
+        id: 'namespace1:resource-one',
+        name: text('Name', 'resource-one'),
+        namespace: text('Namespace', 'namespace1'),
+        date: text('Date', '100 years ago')
+      }
+    ];
+
+    return (
+      <TableComponent
+        title={text('Title', 'Resource Name')}
+        dataRows={dataRows}
+        dataHeaders={[
+          { key: 'name', header: 'Name' },
+          { key: 'namespace', header: 'Namespace' },
+          { key: 'date', header: 'Date Modified' }
+        ]}
+        selectedNamespace="*"
+        loading={boolean('loading', false)}
+        handleNew={action('handleNew')}
+        handleDelete={action('handleDelete')}
+      />
+    );
+  });
+
+  /*
+  HOW TO USE:
+  - Required Props:
+      - title: String name of the Resource being surfaced
+
+      - dataHeaders: Array containing each column (in form of object)
+        appearing in data table.
+        Example:
+          [
+            { key: 'name', header: 'Name' },
+            { key: 'namespace', header: 'Namespace' },
+            { key: 'date', header: 'Date Modified' }
+          ]
+
+      - dataRows: Array containing each row (in form of object)
+        appearing in data table.
+        *all keys in dataHeaders must be specified in each row object*
+          Example:
+            [
+              {
+                **id: 'namespace1:resource-one',
+                name: 'resource-one',
+                namespace: 'namespace1',
+                date: '16 minutes ago'
+              }
+            ]
+          ** Must be specified for React to know which rows have changed,
+            are added, or are removed. It is also used when trying to
+            figure out which row(s) are about to be deleted. Finally, must
+            follow above pattern: {NAMESPACE}:{NAME_OF_RESOURCE} **
+
+      - loading: Boolean Retrieved from Redux store and passed as prop.
+
+      - selectedNamespace: String Retrieved from Redux store and passed as prop.
+        Used when no rows are available and allows us to tell the user in which
+        namespace.
+
+    Optional Props:
+      - HandleNew: Handler used when Add Button is clicked and if not specifiied,
+        Add Button will not appear.
+
+      - HandleDelete: Handler used when Delete Button is clicked and if not
+        specifiied, Delete Button and checkboxes will not appear. Must take 2 arguments,
```

Comment:
If what isn't specified*?

Reply:
if HandleDelete is not specified, the button nor the checkboxes will appear.